src/match.js

```diff
@@ -393,6 +393,8 @@ const checkRecursiveDescendants = (element, path, options, select, toString) =>
         path.unshift(pattern)
         return true
       }
```

Comment:
If matches.length condition goes false, function won't return any value.

Reply:
After discussion with @Anudeep3609 , I implemented a fix here: https://github.com/aptyInc/optimal-select/pull/34 It was also already fixed in 'apty-element-select' in the same way.